### PR TITLE
Feature/gh 648 ansible version

### LINKED ISSUE #648: Upgrade Ansible version from 2.7.9
**As:** a Yorc user
**I want to:** have a more recent Ansible version
**So that:** I can benefit from bugfixes and improvement and provision newer systems like Ubuntu 20.04 for instance which is not properly supported on 2.7.9

**AC1:** Upgrade to a recent Ansible version (newest is better)
**AC2:** Fix deprecated warnings if any (even on bootstrap)

### PR DESCRIPTION
**Pull request by laurentganne**
# Pull Request description

## Description of the change

Using Ansible 2.10.0 (instead of Ansible 2.7.9 in previous version)

### What I did

`commands/bootstrap/resources/topology/tosca_types.zip`:
Fixed deprecation warnings and errors.
See details in Ystia forge pull request https://github.com/ystia/forge/pull/137
 
`commands/bootstrap/setup.go`
Paramiko is not anymore in ansible package requirements, it has to be installed in case the use of openssh is not enforced in yorc configuration.

`prov/ansible/execution.go`
Ansible 2.10 will attempt to discover a suitable Python interpreter on each target host the first time a Python module is executed for that host.
As decribed in https://docs.ansible.com/ansible/latest/reference_appendices/interpreter_discovery.html
this new behavior appears in logs as a warning.
By default here, we set `ansible_python_interpreter` to `auto_silent`, to avoid warnings.

`doc/bootstrap.rst`
Documented the need to install paramiko. Updated examples referencing ansible_python_interpreter

`doc/configuration.rst`
Updated sections referencing ansible_python_interpreter

`doc/install.rst`
Added a command to install paramiko

`doc/tosca.rst`
Updated section on ansible_python_interpreter

`doc/upgrade.rst`

Added a section to describe how to upgrade to ansible 2.10.0

`prov/ansible/yaml_output.go`:
Fixed an issue displaying the output of scripts embedded as ansible shell commands.

`versions.yaml`
Updated ansible version to 2.10.0

### How to verify it

Bootstrap Yorc on gcloud using different target OS versions: CentOS 7, CentOS 8 (fails with Ansible 2.7.9 as this version doesn't detect where python is), Ubuntu 20.04

On the bootrsrapped Yorc, deploy applications made of Compute instances hosting components using ansible playbooks for their interfaces implementations.

Deploy applications using hosted operations as defined in the sample at https://github.com/ystia/tosca-samples/tree/develop/org/ystia/yorc/samples/OrchestratorHost
using these 2 different Yorc configurations:
* unsandboxed operations allowed
* sandbox, using for example the docker image jfloff/alpine-python:2.7-slim

### Description for the changelog

Upgrade Ansible version from 2.7.9 ([GH-648](https://github.com/ystia/yorc/issues/648))

## Applicable Issues

Fixes #648

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +17,7 @@
 
 ### ENHANCEMENTS
 
+* Upgrade Ansible version from 2.7.9 ([GH-648](https://github.com/ystia/yorc/issues/648))
```
Comment (loicalbertin) [inline CHANGELOG.md:20]:
```suggestion
* Upgrade Ansible version from 2.7.9 to 2.10.0 ([GH-648](https://github.com/ystia/yorc/issues/648))
```

File: commands/bootstrap/setup.go [hunk 2]
```diff
@@ -546,6 +551,22 @@ func installAnsible(version string) error {
...
+// installParamiko installs Paramiko needed when openSSH is not used
+// Install using pip3 if available else using pip
+func installParamiko() error {
+	pipCmd, err := getPipCmd()
+	if err != nil {
+		return err
+	}
+
+	// Installing ansible
+	cmd := exec.Command(pipCmd, "install", "--upgrade", "paramiko")
```
Comment (loicalbertin) [inline commands/bootstrap/setup.go:563]: As per [comment on Forge PR](https://github.com/ystia/forge/pull/137#discussion_r501508861) should we use a specific or bounded version for paramiko?

**Review by loicalbertin — APPROVED** [review]
LGTM :+1: :clap:

File: CHANGELOG.md (hunk 1 above)
Comment (laurentganne) [reply CHANGELOG.md]: ok, updated

File: commands/bootstrap/setup.go (hunk 2 above)
Comment (laurentganne) [reply commands/bootstrap/setup.go:563]: As the Ansible doc doesn't specify any specific or bounded version, proposing to let it like this

Comment (loicalbertin) [reply commands/bootstrap/setup.go:563]: ok

**Pull request merged**